### PR TITLE
3-get-api

### PR DESCRIPTION
**Pull request by gilaineyo**
Adds GET /api to respond with list of endpoints, and test. package.json updated to add fs/promises

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -26,6 +26,23 @@ describe('/api/topics', () => {
...
+        .get('/api')
+        .expect(200)
+        .then(({body}) => {
+            const { endpoints } = body
+            expect(typeof endpoints).toBe('object')
+            expect(Array.isArray(endpoints)).toBe(false)
+            expect(endpoints).toMatchObject({
+                "GET /api": { "description": expect.any(String) },
+                "GET /api/topics": { "description": expect.any(String), "queries": [], "exampleResponse": { "topics": [{ "slug": expect.any(String), "description": expect.any(String) }] }},
+            })
```
Comment (Keb84) [inline __tests__/app.test.js]: If you require in the endpoints file you can then expect that the file will always look like whatever is in it at the time. As you'll be adding to the file as you go along with all of your endpoints you want the test to be dynamic.

**Review by Keb84 — CHANGES_REQUESTED** [review]
Nice work 🤸 please have a look at my comment, make the change and resubmit 🙂

File: __tests__/app.test.js (hunk 1 above)
Comment (gilaineyo) [reply __tests__/app.test.js]: Thanks Kirsty, I'll have a look when I've committed on my current branch. So am I right in thinking I should be referencing the file that I've required in in the tests? Is the test really just "matches the endpoints file" and if that file's wrong there's not really a way to test for that?

**Review by Keb84 — APPROVED** [review]
Nice work. Merge when ready 🙂

**Pull request merged**